File: app/src/main/java/com/tehran/motivation/util/AddCookiesInterceptor.java
package com.tehran.motivation.util;

import android.content.Context;
import android.preference.PreferenceManager;

import java.io.IOException;
import java.util.HashSet;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;
import timber.log.Timber;

public class AddCookiesInterceptor implements Interceptor {

    public static final String PREF_COOKIES = "PREF_COOKIES";
    private Context context;

    public AddCookiesInterceptor(Context context) {
        this.context = context;
    }

    @Override
    public Response intercept(Interceptor.Chain chain) throws IOException {
        Request.Builder builder = chain.request().newBuilder();

        HashSet<String> preferences = (HashSet<String>) PreferenceManager.getDefaultSharedPreferences(context).getStringSet(PREF_COOKIES, new HashSet<>());
        String cookiestring = "";
        for (String cookie : preferences) {
            String[] parser = cookie.split(";");
            cookiestring = cookiestring + parser[0] + "; ";
        }
        builder.addHeader("Cookie", cookiestring);
        return chain.proceed(builder.build());
    }
}

File: app/src/main/java/com/tehran/motivation/util/AddCookiesInterceptor.kt
package com.tehran.motivation.util

import android.content.Context
import androidx.preference.PreferenceManager
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException
import java.util.*
import kotlin.jvm.Throws

class AddCookiesInterceptor(private val context: Context) : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()
        val preferences =
            PreferenceManager.getDefaultSharedPreferences(context).getStringSet(
                PREF_COOKIES,
                HashSet()
            ) as HashSet<String>
        var cookiestring = ""
        for (cookie in preferences) {
            val parser = cookie.split(";".toRegex()).toTypedArray()
            cookiestring = cookiestring + parser[0] + "; "
        }
        builder.addHeader("Cookie", cookiestring)
        return chain.proceed(builder.build())
    }

    companion object {
        const val PREF_COOKIES = "PREF_COOKIES"
    }
}